docs/design-system.md

```diff
@@ -0,0 +1,73 @@
+3. Subdirectory called `helpers` - these are styles meant to be quick references for one off components in the editor and dashboard that aren’t in the design system.They’re also used in the design system for continuity.
+4. Subdirectory called `global` - these are exported as `ThemeGlobals` and should be added to app root code to handle global styles for both the dashboard and editor. Currently only the focus override exists here.
+5. Subdirectory called `constants` - these are constants that are used in both the dashboard and the editor (or needed also in the design system) so that we can have a single source of truth. You can import them via `THEME _CONSTANTS`. These include things like static widths of wordPress menus that all projects reference.
+
+It’s important to note that this is a living resource - I fully anticipate that we need to add to the theme as we begin to use it more robustly. The design system is being created iteratively as we go through implementing new designs. If there’s a new need for a helper, add it! If there’s a color lacking, add it in following the existing patterns. My goal here was to create a base that we can easily keep updating and avoid needless duplication.
+
+## Components
+
+The hope here is to make the transition to using the components in the design system straightforward and easy. If a component in the design system already existed in the editor or dashboard, that source code was used as a reference for the new one in almost all cases. Props should be the same or similar for components that were based off of existing code.
+
+### Differences from the editor
+
+- Icon buttons are now a button type where you give the icon to the button instead of specific button components.
+- Button refs are readily available
+- Typography is available in molecules - there’s `Display`, `Headline`, `Text`, and `Link` that match the figma specs for type and can take in a styled component `as` of `forwardedAs` to keep HTML semantic but avoid needlessly styling each text element. If you do need to do that there’s a `themeHelpers.expandPresetStyles` or `themehelpers.expandTextPreset` you can look to.
+
+### How to add components to the design system
+
+There’s going to be things missing from the design system as of this writing. Our designs have been in flux and we’ve been adding things as we need them in the redesign. That said, feel free to add missing components. Components live in [design-system/components](https://github.com/google/web-stories-wp/tree/main/assets/src/design-system/components). This directory’s pretty flat so that things are easy to find at 1 level.  The general rule here, as with any shared design system, is that we keep things generic and non-opinionated as much as possible so that the usability remains flexible. All components have storybook demos that allow all props passed to the component to be seen as knobs as well as dark and light versions so that theme colors can be checked.
```

Comment:
Is `there's` a contraction for `there are`? I think I usually say "there's" in this context, but reading it aloud makes me doubt myself.
```suggestion
There are going to be things missing from the design system. Designs have been in flux and we’ve been adding things as we need them in the redesign. That said, feel free to add missing components. Components live in [design-system/components](https://github.com/google/web-stories-wp/tree/main/assets/src/design-system/components). This directory’s pretty flat so that things are easy to find at 1 level.  The general rule here, as with any shared design system, is that we keep things generic and non-opinionated as much as possible so that the usability remains flexible. All components have storybook demos that allow all props passed to the component to be seen as knobs as well as dark and light versions so that theme colors can be checked.
```